src/easy_mark_parser.rs

```diff
@@ -0,0 +1,342 @@
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Item<'a> {
+    /// `\n`
+    Newline,
+    ///
+    Text(Style, &'a str),
+    /// title, url
+    Hyperlink(Style, &'a str, &'a str),
+    /// leading space before e.g. a [`Self::BulletPoint`].
+    Indentation(usize),
+    /// >
+    QuoteIndent,
+    /// - a point well made.
+    BulletPoint,
+    /// 1. numbered list. The string is the number(s).
+    NumberedPoint(&'a str),
+    /// ---
+    Separator,
+    /// language, code
+    CodeBlock(&'a str, &'a str),
+}
+
+#[derive(Copy, Clone, Debug, Default, Eq, PartialEq)]
+pub struct Style {
+    /// # heading (large text)
+    pub heading: bool,
+    /// > quoted (slightly dimmer color or other font style)
+    pub quoted: bool,
+    /// `code` (monospace, some other color)
+    pub code: bool,
+    /// self.strong* (emphasized, e.g. bold)
+    pub strong: bool,
+    /// _underline_
+    pub underline: bool,
+    /// ~strikethrough~
+    pub strikethrough: bool,
+    /// /italics/
+    pub italics: bool,
+    /// $small$
+    pub small: bool,
+    /// ^raised^
+    pub raised: bool,
+}
+
+/// Parser for the `EasyMark` markup language.
+///
+/// See the module-level documentation for details.
+///
+/// # Example:
+/// ```
+/// # use egui_demo_lib::easy_mark::parser::Parser;
+/// for item in Parser::new("Hello *world*!") {
+/// }
+///
+/// ```
+pub struct Parser<'a> {
```

Comment:
testy, testy, testy, to się aż prosi o testy! Zróbcie testy koniecznie w drugiej części!